python/lsst/afw/cameraGeom/camera.py

```diff
@@ -68,118 +73,73 @@ def getPupilFactory(self, visitInfo, pupilSize, npix, **kwargs):
-            return detTrans.transform(positionList, fromSys, toSys)
+            det = self[fromSys.getDetectorName()]
+            return det.getTransformMap().getTransform(fromSys, toSys)
         elif toSys.hasDetectorName():
             det = self[toSys.getDetectorName()]
-            detTrans = det.getTransformMap()
-            return detTrans.transform(positionList, fromSys, toSys)
-        elif toSys in self._transformMap:
-            # use camera transform map
-            return self._transformMap.transform(positionList, fromSys, toSys)
-        raise RuntimeError(
-            "Could not find mapping from %s to %s"%(fromSys, toSys))
-
-    def _transformSingleSys(self, cameraPoint, toSys):
-        """!Transform a CameraPoint with a CameraSys to another CameraSys.
+            return det.getTransformMap().getTransform(fromSys, toSys)
+        else:
+            return self.getTransformMap().getTransform(fromSys, toSys)
 
-        @warning This method only handles a single jump, not a transform linked by a common native sys.
+    def findDetectors(self, point, cameraSys):
+        """!Find the detectors that cover a point in any camera system
 
-        @param[in] cameraPoint  CameraPoint to transform
-        @param[in] toSys  Destination coordinate system
-        """
-        fromSys = cameraPoint.getCameraSys()
-        if fromSys.hasDetectorName():
-            # use from detector to transform
-            det = self[fromSys.getDetectorName()]
-            return det.transform(cameraPoint, toSys)
-        elif toSys.hasDetectorName():
-            # use the to detector to transform
-            det = self[toSys.getDetectorName()]
-            return det.transform(cameraPoint, toSys)
-        elif toSys in self._transformMap:
-            # use camera transform map
-            outPoint = self._transformMap.transform(
-                cameraPoint.getPoint(), cameraPoint.getCameraSys(), toSys)
-            return CameraPoint(outPoint, toSys)
-        raise RuntimeError("Could not find mapping from %s to %s" %
-                           (cameraPoint.getCameraSys(), toSys))
-
-    def findDetectors(self, cameraPoint):
-        """!Find the detectors that cover a given cameraPoint, or empty list
-
-        @param[in] cameraPoint  position to use in lookup
+        @param[in] point  position to use in lookup (lsst.afw.geom.Point2D)
+        @param[in] cameraSys  camera coordinate system of `point`
         @return a list of zero or more Detectors that overlap the specified point
         """
-        # first convert to focalPlane since the point may be in another
-        # overlapping detector
-        nativePoint = self._transformSingleSys(
-            cameraPoint, self._nativeCameraSys)
+        # convert `point` to the native coordinate system
+        transform = self._getTransformFromOneTransformMap(cameraSys, self._nativeCameraSys)
+        nativePoint = transform.applyForward(point)
 
         detectorList = []
```

Comment:
Because a point *can* be on multiple detectors. I don't believe any instruments presently use that capability, but an imaging instrument with a beamsplitter certainly would.